csrc/codegen.cpp

```diff
@@ -2810,35 +2815,41 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
                     ->buffer()
                     ->isA<TensorView>());
 
-      for (const auto& group_index : arange(index_replacement_maps.size())) {
-        // Set the index replacement map with the concrete values of
-        // indices of grouped loops.
-        index_replacement_map_ = index_replacement_maps.at(group_index);
```

Comment:
This part doesn't do `T1.array` but it's more flexible as it actually passes each individual element rather than passing one contiguous buffer. While that's certainly more flexible, I'm not sure if it's really necessary and maybe just over engineering. In this PR, I changed it to pass just a contiguous buffer much like the other grouped calls.